### PR TITLE
refactor(demo) more hook and storybook

### PR DESCRIPTION
**Pull request by guillaumechervetaxa**

### CODE REVIEW

File: examples/demo/src/shared/Title/Title.js [hunk 1]
```diff
@@ -3,13 +3,13 @@ import { Link } from 'react-router-dom';
 import { PropTypes } from 'prop-types';
 import './Title.scss';
 
-const TitleBar = ({ backHome, children, className }) => (
+const TitleBar = ({ onClick, children, className }) => (
   <div className={className}>
     <div className="container-fluid container af-title-bar__wrapper">
-      {backHome && (
-        <Link title="Retour à l'accueil" className="btn af-btn--circle" to="/">
```
Comment (samuel-gomez-axa) [inline examples/demo/src/shared/Title/Title.js]: why you don't use Link router component ?

File: examples/demo/src/withCustomFetch.js [hunk 2]
```diff
@@ -3,17 +3,17 @@ import { withEnvironment } from './EnvironmentProvider';
...
-  const url = apiBaseUrl.replace("{path}", path)
+  const url = apiBaseUrl.replace("{path}", path);
   const response = await fetch(url, config);
   return response.json();
 };
 
-const withCustomFetch = fetch => Component => props => {
-  console.log(fetch,props.environment, Component );
+const withCustomFetch = fetch => Component => ({environment, ...otherProps}) => {
+  console.log(fetch, environment, Component );
```
Comment (samuel-gomez-axa) [inline examples/demo/src/withCustomFetch.js]: Should be delete

File: examples/demo/src/shared/Title/Title.js (hunk 1 above)
Comment (guillaumechervetaxa) [reply examples/demo/src/shared/Title/Title.js]: It was not working inside my stories ^^
I generaly Never use it.

File: examples/demo/src/Home/Home.container.js [hunk 3]
```diff
@@ -4,30 +4,33 @@ import fetchDevis from './Home.service';
...
+  const onClick = (e) => {
+    e.preventDefault();
+    history.push('/new');
+  };
 
+  return (<HomeWithLoader {...state} onClick={onClick} />);
 };
 
-export default withCustomFetch(fetch)(HomeContainer);
+export default withCustomFetch(fetch)(withRouter(HomeContainer));
```
Comment (samuel-gomez-axa) [inline examples/demo/src/Home/Home.container.js]: composition should be more readable

File: examples/demo/src/shared/Title/Title.js (hunk 1 above)
Comment (samuel-gomez-axa) [reply examples/demo/src/shared/Title/Title.js]: you just have to wrap your story by a Router
i think you also create global decorator

**Pull request merged**